adaptors/collections.md

```diff
@@ -0,0 +1,200 @@
+---
+title: Collections Adaptor
+---
+
+## Collections Overview
+
+The Collections API is a key/value storage solution. It is designed for high
+performance over a large volume of data.
+
+Use-cases include:
+
+- Storing mapping objects for use in workflows
+- Buffering and aggregating high volumes of incoming data
+- Caching and sharing state between workflows
+
+A Collection is bound to a project. Collections can only be accessed with a
+token associated with that project. When running on the app, a workflow is
+automatically granted access to all collections on the same project. When
+running in the CLI, a Personal Access Token can be used (generated from the app
+at /profile/tokens).
+
+## The Collections Adaptor
+
+The Collections adaptor is a special adaptor. Uniquely, the Collections adaptor
+is designed to be run _alongside_ other adaptors, and is injected for you by the
+platform.
+
+This makes the Collections API available to every step in a workflow, regardless
+of which adaptor it is using.
+
+## Usage Guide
+
+### Set some data in a collection
+
+The Collection API allows you to set a JSON object (or any primitive JS value)
+under a given key:
+
+You can also pass an array of items for a batch-set.
+
+### Getting data from a collection
+
+To retrieving multiple items from a Collection, we recommend using the `each()`
+function.
+
+`each()` will stream each value individually, greatly reducing the memory
```

Comment:
Do we really have to use `each()`? What about the scenario when I want to query a list of devices that match my search criteria? In some of the use cases we've been imaging so far, we've been getting a long list of records and then sending that entire list to a downstream system, or comparing that list with data from another system. Mtuchi could show you what's going on the openlmis prototype, if interested in learning more.

Reply:
I see `get()` now and left a comment below